neqo-http3/src/server_events.rs

```diff
@@ -53,24 +78,22 @@ impl ClientRequestStream {
     /// It may return `InvalidStreamId` if a stream does not exist anymore.
     pub fn set_response(&mut self, headers: &[Header], data: &[u8]) -> Res<()> {
         qinfo!([self], "Set new response.");
-        self.handler
-            .borrow_mut()
-            .set_response(self.stream_id, headers, data)
+        self.stream_handler.handler.borrow_mut().set_response(
+            self.stream_handler.stream_id,
+            headers,
+            data,
+        )
     }
 
     /// Request a peer to stop sending a request.
     /// # Errors
     /// It may return `InvalidStreamId` if a stream does not exist anymore.
     pub fn stream_stop_sending(&mut self, app_error: AppError) -> Res<()> {
-        qdebug!(
-            [self],
-            "stop sending stream_id:{} error:{}.",
-            self.stream_id,
-            app_error
-        );
-        self.conn
+        qdebug!([self], "stop sending error={}.", app_error);
+        self.stream_handler
+            .conn
             .borrow_mut()
-            .stream_stop_sending(self.stream_id, app_error)?;
+            .stream_stop_sending(self.stream_handler.stream_id, app_error)?;
```

Comment:
Please move some of this to the new struct, particularly these parts that are generic.